snowpack/src/build/build-pipeline.ts

```diff
@@ -140,36 +154,28 @@ async function composeSourceMaps(
  */
 async function runPipelineTransformStep(
   output: SnowpackBuildMap,
-  srcPath: string,
   {isDev, plugins, sourceMaps}: BuildFileOptions,
 ): Promise<SnowpackBuildMap> {
-  const srcExt = getExt(srcPath).baseExt;
-  const rootFilePath = srcPath.replace(srcExt, '');
-  const rootFileName = path.basename(rootFilePath);
   for (const step of plugins) {
     if (!step.transform) {
       continue;
     }
-
     try {
       for (const destExt of Object.keys(output)) {
-        const destBuildFile = output[destExt];
-        const {code} = destBuildFile;
-        const fileName = rootFileName + destExt;
-        const filePath = rootFilePath + destExt;
-        const debugPath = path.relative(process.cwd(), filePath);
-        logger.debug(`transform() starting… [${debugPath}]`, {name: step.name});
+        const {code, outPath} = output[destExt];
+        const filePath = path.basename(outPath);
+        logger.debug(`transform() starting… [${outPath}]`, {name: step.name});
         const result = await step.transform({
           contents: code,
           isDev,
           fileExt: destExt,
-          id: filePath,
+          id: outPath,
           // @ts-ignore: Deprecated
-          filePath: fileName,
+          filePath: filePath,
```

Comment:
this seems to be changing a lot of existing behavior to use new values. our entire plugin ecosystem relies on these values not changing